pkgs/default.nix

```diff
@@ -46,16 +50,26 @@
           pytest-output-to-files
           ;
       };
+      # use libre-soc's fork of pyelftools, few commits have diverged from pyelftools packaged by nixpkgs
       libresoc-pyelftools = callPackage ./libresoc/libresoc-pyelftools.nix {};
       mdis = callPackage ./libresoc/mdis.nix {};
+      # upstream nmigen has been renamed to https://github.com/amaranth-lang/amaranth.
+      # amaranth is packaged in nixpkgs but we can't just override a few of the attributes the way we did for pyelftools,
+      # because the names are different, so much of this is copied from the amaranth build recipe.
       nmigen = callPackage ./libresoc/nmigen.nix {};
+      # nmigen-soc was not currently packaged in nixpkgs during work on this PR. However,
+      # it was also renamed to https://github.com/amaranth-lang/amaranth-soc and has now been packaged in nixpkgs,
+      # but suffers from the same rename issue as the previous commit with nmigen and amaranth
       nmigen-soc = callPackage ./libresoc/nmigen-soc.nix {inherit nmigen;};
+      # upstream tool for running verilog circuit against linux emulator
       nmutil = callPackage ./libresoc/nmutil.nix {inherit nmigen pytest-output-to-files;};
+      # upstream library for running pin multiplexers
       pinmux = callPackage ./libresoc/pinmux.nix {};
+      # upstream emulator of power ISA
       power-instruction-analyzer = callPackage ./libresoc/power-instruction-analyzer.nix {};
+      # upstream test harness
```

Comment:
I think we discussed this during the call but we were a bit inconsistent about what upstream meant, how about here we `s/upstream/libresoc` if the project is a libresoc specific project (unlike nmigen/amaranth which has its own upstream independent of libresoc)